avatars/avatars.js

```diff
@@ -2652,6 +2658,218 @@ class Avatar {
       }
     };
     _updateEyeTarget();
+
+    const _updateEyeballTarget = () => {
+      const leftEye = this.modelBoneOutputs['Eye_L'];
```

Comment:
For this feature, getting the bones out of modelBoneOutputs is the main connection to avatars.js, would it be worth splitting this out?  and having a function that returns the bones to anything that requires it,  and that external object can handle the eye target updates for mirrors

Reply:
> For this feature, getting the bones out of modelBoneOutputs is the main connection to avatars.js, would it be worth splitting this out? and having a function that returns the bones to anything that requires it, and that external object can handle the eye target updates for mirrors

Yes, I agree with that. I actually try to follow this design pattern with `Player`, which has e.g. sfx, physics, and the avatar separated out.

However, I think this issue is not isolated to the eyeball handling. Pretty much the entire `avatars.js` update loop can be broken out like this. The codebase is intentionally architected with strong logic isolation to facilitate these break-outs any time, so that "cleaning up" the code is mostly just moving the file it resides in. That is the point `_updateEyeballTarget`.

IMO we should clean up all of `avatars.js`, in a separate PR.